server/src/Q3Server/AuthParameterChecker.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Owin;
+
+namespace Q3Server
+{
+    public class AuthParameterChecker : OwinMiddleware
+    {
+        public AuthParameterChecker(OwinMiddleware next) : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (HasCorrectParameterCombination(context.Request))
+            {
+                await Next.Invoke(context);
+            }
+            else
+            {
+                context.Response.StatusCode = 400;

```

Comment:
I think this is right - eg https://www.scottbrady91.com/Katana/Creating-OWIN-Middleware-using-Microsoft-Katana